community/bolt/src/main/java/org/neo4j/bolt/security/auth/BasicAuthentication.java

```diff
@@ -70,9 +70,19 @@ public AuthenticationResult authenticate( Map<String,Object> authToken ) throws
         }
     }
 
+    @Override
+    public void logout( AccessMode accessMode )
+    {
+        if ( accessMode instanceof AuthSubject )
```

Comment:
It feels like somewhat of a design flaw that we need to do these casts. Could we have something along the line of
```
public interface Authentication<T extends AccessMode>
{
...
    public void logout( T accessMode );
}
```
and then have `BasicAuthentication implements Authentication<AuthSubject>`